lib/releases.sh

```diff
@@ -1,44 +1,20 @@
 #!/bin/bash
 
 function get_latest_release() {
-  # get last page
+  # set url to get all releases from github
+  url="${1}?per_page=1000"
+  
+  # This gets the latest release as vx.y.z or vx.y.z-rc.0, including any version with a suffix starting with - , for example -rc.0
+  # The order is exactly as released in Github.
+  # Downside is that selecting official releases only isn't possible, while pre-release
+  # selection is possible given specific enough prefix, like v1.3.0-pre
+
   if [ -z "${GITHUB_TOKEN:-}" ]; then
-    last_page="$(curl -s -I "${1}" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g')"
+        release_tag=$(curl -sL "${url}" | jq .[].name -r | grep -E "${2:-}" -m 1)
```

Comment:
You want to fail if `$2` is empty, since then the caller has not passed the release to search for.

Just add to top of the function:
`local release="${2:?no release given}"` and it'll fail nicely if its not passed, then use `${release}` in grep.